### ISSUE TITLE
Liveness Detection

### ISSUE THREAD

**Issue by GL-RuiLa** [justadudewhohacks/face-api.js#221]
Hi,

I am working on an internal room check-in system using this API. The detection speed and accuracy looks great. But there is a problem that people can use pictures on their phone or printed paper to unlock the door. I am wondering if there is any liveness detection (anti-spoofing) model we can integrate to make this API better. Thanks 

Rio

**Comment by nathany** [comment]
Also interested in liveness detection.

One aspect of that could be blink detection, see #176.

**Comment by leandrojo** [comment]
Also interested! Something like that!
http://cvlab.cse.msu.edu/project-face-anti.html

**Comment by nathany** [comment]
An article on one type of liveness detection using a convolutional neural network: https://www.pyimagesearch.com/2019/03/11/liveness-detection-with-opencv/

**Comment by guinaldi** [comment]
Also Interested!

**Comment by Bajajar** [comment]
I would love to see Vincent @justadudewhohacks implement some kind like this https://github.com/richmondu/libfaceid

**Comment by ScottDellinger** [comment]
This is likely my #1 request for this fantastic library.

**Comment by ariffromeo** [comment]
I have converted liveness keras model from [https://www.pyimagesearch.com/2019/03/11/liveness-detection-with-opencv/](https://www.pyimagesearch.com/2019/03/11/liveness-detection-with-opencv/) into tensorflowjs json model. anyone know how can i use this converted model in this library?

![live](https://user-images.githubusercontent.com/8829689/66623410-1e1e7f80-ec1e-11e9-8861-2a29b3f6c405.JPG)

**Comment by punisher97** [comment]
New open source solution: https://github.com/FaceGg/FeatherNets_Face-Anti-spoofing-Attack-Detection-Challenge-CVPR2019

**Comment by ankitnamdeo34** [comment]
> An article on one type of liveness detection using a convolutional neural network: https://www.pyimagesearch.com/2019/03/11/liveness-detection-with-opencv/

I tried that but the accuracy of the model is very poor.

**Comment by startupgurukul** [comment]
Landmarks if they are accurate, specifically for eyes, changes in difference of distance between upper eye points and lower eyes points we tried to do liveliness detection using the same but sadly eyes closed want perfoming well for Indians faces

**Comment by Cheyenne55** [comment]
Hello there,

Just wanted to suggest a direction on how to do this.

There's a company which I won't disclose the name, who does the liveness detection based on the realness of the skin of the person.

While capturing the video input, several screens of different colors are quickly displayed on the screen. The algorithm analyses the reflection of light on the skin and determines if it's real or not.

Trib.

**Comment by sundar505** [comment]
I want code example for checking face liveness.

**Comment by tuanphan1998** [comment]
Hopefully this repo has an additional guide to detecting fake faces

**Comment by cristian2395** [comment]
@ariffromeo, sorry, did you manage to incorporate the models?

**Comment by crypt0miester** [comment]
excellent feature request. would love to have this.

**Comment by financeable** [comment]
+100 on this request!

**Comment by Turing311** [comment]
Check this repo
https://github.com/Nikit333/Face-Liveness-Detection

**Comment by Rdp1111** [comment]
Please check following repo:
https://github.com/Nikit333/Face-Liveness-Detection

**Comment by qalainau** [comment]
I think this code will work.
https://gist.github.com/kleysonr/d75494f239ad0dce561a55a624920693

**Comment by Alireza17224** [comment]
I have a problem like this too. Please fix it 😍😍 .

**Comment by codevjs** [comment]
+1
